github/retry.go

```diff
@@ -29,28 +24,15 @@ func (t *retryTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	var err error
 	retry(func() error {
 		r, err = t.T.RoundTrip(req)
-		if err != nil {
-			if err == context.Canceled {
-				return &errUnretriable{Err: err}
-			}
-
-			return err
-		}
-
-		if r.StatusCode == http.StatusOK {
+		if err == context.Canceled {
 			return nil
 		}
 
-		body, _ := ioutil.ReadAll(r.Body)
-
-		// Restore the io.ReadCloser
-		r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
-
-		err = fmt.Errorf("non-200 OK status code: %v body: %q", r.Status, body)
-		if r.StatusCode > 500 {
+		if err != nil || r.StatusCode >= 500 {
```

Comment:
`4xx` errors should not be retried, only errors got from `RoundTrip` and `5xx`

Reply:
most probably Marvin also failed to understand this code after the change :)

Reply:
@smacker I just read your review, and yup I completely missed that `err` is correct because it's defined outside 😂. I also think that before was way clearer, and the backoff library uses the same approach of having unretriable errors.